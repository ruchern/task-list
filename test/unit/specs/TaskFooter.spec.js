import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import TaskFooter from '@/components/TaskFooter'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TaskFooter', () => {
  let wrapper
  let store, state, mutations

  beforeEach(() => {
    state = {
      tasks: [
        { name: 'First task', completed: true },
        { name: 'Second task', completed: false }
      ],
      query: ''
    }

    mutations = {
      setFilter: jest.fn()
    }

    store = new Vuex.Store({
      state,
      mutations
    })

    wrapper = shallowMount(TaskFooter, {
      attachToDocument: true,
      store,
      localVue
    })
  })

  it('should render the component', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should set the filter based on the value it was clicked', () => {
    wrapper.find('span').trigger('click')
    expect(mutations.setFilter).toHaveBeenCalled()
  })
})
