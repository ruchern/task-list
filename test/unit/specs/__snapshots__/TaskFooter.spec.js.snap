// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskFooter should render the component 1`] = `
<footer
  class="card-footer"
>
  <div
    class="card-footer-item"
  >
    <small>
      1 task left
    </small>
  </div>
   
  <div
    class="card-footer-item"
  >
    <div
      class="tags"
    >
      <span
        class="tag is-link"
      >
        
        All
      
      </span>
      <span
        class="tag is-link"
      >
        
        Completed
      
      </span>
      <span
        class="tag is-link"
      >
        
        Todo
      
      </span>
    </div>
  </div>
</footer>
`;
